src/app/public/modules/autocomplete/autocomplete.component.spec.ts

```diff
@@ -402,6 +403,28 @@ describe('Autocomplete component', () => {
 
       expect(rendererSpy).toHaveBeenCalledWith(dropdownElement, 'width', formattedWidth);
     }));
+
+    it('should search after debounce time', fakeAsync(() => {
+      component.debounceTime = 400;
+      fixture.detectChanges();
+
+      const inputElement = getInputElement();
+      const spy = spyOn(autocomplete, 'search').and.callThrough();
+
+      inputElement.value = 'r';
+      SkyAppTestUtility.fireDomEvent(inputElement, 'keyup');
+
+      // The search method should not execute at this time.
+      tick(10);
+      expect(spy).not.toHaveBeenCalled();
```

Comment:
I notice you aren't using `fixture.detectChanges()` in the `tick(10)` scenario, but you do use in in the`tick(400)` scenario. Does this impact the test? Just trying to make sure we're comparing apples and apples and not getting a false positive.

Reply:
Good eye. I didn't put a `detectChanges` there because there wasn't any data binding taking place. But I threw one in there just in case.